tests/unit_tests/hyperion/external_interaction/test_agamemnon.py

```diff
@@ -0,0 +1,177 @@
+import json
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from mx_bluesky.common.parameters.constants import GridscanParamConstants
+from mx_bluesky.hyperion.external_interaction.agamemnon import (
+    PinType,
+    SinglePin,
+    _get_pin_type_from_agamemnon_parameters,
+    get_next_instruction,
+    get_pin_type_from_agamemnon,
+    update_params_from_agamemnon,
+)
+from mx_bluesky.hyperion.parameters.load_centre_collect import LoadCentreCollect
+
+
+@pytest.mark.parametrize(
+    "num_wells, well_width, buffer, expected_width",
+    [
+        (3, 500, 0, 1000),
+        (6, 50, 100, 450),
+        (2, 800, 50, 900),
+    ],
+)
+def test_given_various_pin_formats_then_pin_width_as_expected(
+    num_wells, well_width, buffer, expected_width
+):
+    pin = PinType(num_wells, well_width, buffer)
+    assert pin.full_width == expected_width
+
+
+def params_from_loop_type(loop_type: str | None):
+    return {"sample": {"loopType": loop_type}}
+
+
+def test_given_no_loop_type_in_parameters_then_single_pin_returned():
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(None))
+        == SinglePin()
+    )
+
+
+@pytest.mark.parametrize(
+    "loop_name, expected_loop",
+    [
+        ("multipin_6x50+9", PinType(6, 50, 9)),
+        ("multipin_6x25.8+8.6", PinType(6, 25.8, 8.6)),
+        ("multipin_9x31+90", PinType(9, 31, 90)),
+    ],
+)
+def test_given_multipin_loop_type_in_parameters_then_expected_pin_returned(
+    loop_name: str, expected_loop: PinType
+):
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+        == expected_loop
+    )
+
+
+@pytest.mark.parametrize(
+    "loop_name",
+    [
+        "nonesense",
+        "single_pin_78x89+1",
+    ],
+)
+@patch("mx_bluesky.hyperion.external_interaction.agamemnon.LOGGER")
+def test_given_completely_unrecognised_loop_type_in_parameters_then_warning_logged_single_pin_returned(
+    mock_logger: MagicMock,
+    loop_name: str,
+):
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+        == SinglePin()
+    )
+    mock_logger.warning.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    "loop_name",
+    [
+        "multipin_67x56",
+        "multipin_90+4",
```

Comment:
Following my previous comment, these invalid pin types can be tests with a stricted regex, but again not sure how important this really is:
```suggestion
        "multipin_67x56",
        "multipin_90+4",
        "multipin_8",
        "multipin_6x50+",
        "multipin_6x50+98.",
        "multipin_6x50+.1",
        "multipin_6x.50+98",
        "multipin_6x50+98.1.2",
        "multipin_6x50.5.6+98",
        "multipin_6x50+98..1",
        "multipin_6x.50+.98",
        "multipin_6x+98",
```